app/controllers/spree/stripe_controller.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Spree
+  class StripeController < Spree::BaseController
+    include Core::ControllerHelpers::Order
+
+    def confirm_payment
+      begin
+        if params[:payment_method_id].present?
+          # Create the PaymentIntent
+          intent = stripe.create_intent(
+            (current_order.total * 100).to_i,
+            params[:payment_method_id],
+            currency: current_order.currency,
+            confirmation_method: 'manual',
+            confirm: true,
+            setup_future_usage: 'on_session',
+            metadata: { order_id: current_order.id }
+          )
+        elsif params[:payment_intent_id].present?
+
+          intent = stripe.confirm_intent(params[:payment_intent_id], nil)
+        end
+      rescue Stripe::CardError => e
+        # Display error on client
+        render json: { error: e.message }
+        return
+      end
+
+      generate_payment_response(intent)
+    end
+
+    private
+
+    def stripe
+      @stripe ||= Spree::PaymentMethod::StripeCreditCard.first
```

Comment:
I'd love to find an alternative way for this. I saw the `payment_method_id` that we could use but maybe it's not always available?

Reply:
We need to access that record in the controller because its class encapsulates all Stripe related knowledge, the previous one and the new one introduced with this PR. I can't think about a way to do things differently without changing much more code... do you have any suggestion?

Reply:
Does it need to be taken from the database? Just thinking aloud here, but maybe we can do `Spree::PaymentMethod::StripeCreditCard.new` and not relying on having the payment method saved? Just wondering if it can create issues if a store has two stripe payment methods on the db for some reason.

Reply:
@kennyadsl with the help of @DanielePalombo I understood the issue you exposed. It's now fixed by loading the right payment method dynamically.